src/util/util_creds.h

```diff
@@ -70,14 +70,15 @@ struct cli_creds {
     SELINUX_CTX selinux_ctx;
 };
 
-#define cli_creds_get_uid(x) x->ucred.uid
-#define cli_creds_get_gid(x) x->ucred.gid
+#define cli_creds_get_uid(x) (x->ucred.uid)
+#define cli_creds_get_gid(x) (x->ucred.gid)
 
 #else /* not HAVE_UCRED */
 struct cli_creds {
     SELINUX_CTX selinux_ctx;
 };
-#define cli_creds_get_uid(x) -1
+#define cli_creds_get_uid(x) (-1)
```

Comment:
This is more of a theoretical question than anything else. Is there any system out there that doesn't support ucred? I'm asking because we never check the return value of `cli_creds_get_uid()` and in a very hypothetical and strange scenario returning -1 may match with an existing id (UID_MAX - 1).

Reply:
I'm all for hard fail at ./configure time if UCRED isn't supported:
https://github.com/SSSD/sssd/blob/8218634df52dd33524e724e78dded2deb1aa283e/src/external/platform.m4#L47

It is much easier and makes more sense than trying to handle (-1).

But this is out of scope for this patch.